packages/neos-ui/src/manifest.dataloaders.js

```diff
@@ -0,0 +1,160 @@
+import HLRU from 'hashlru';
+import manifest from '@neos-project/neos-ui-extensibility';
+import {SynchronousRegistry} from '@neos-project/neos-ui-extensibility/src/registry';
+import backend from '@neos-project/neos-ui-backend-connector';
+
+function makeCacheKey(prefix, params) {
+    return prefix + JSON.stringify(params);
+}
+
+manifest('main.dataloaders', {}, globalRegistry => {
+    //
+    // Create container registry
+    //
+    globalRegistry.add('dataLoaders', new SynchronousRegistry(`
+        # A "Data Loader" controls asynchronous loading of secondary data, which is used in all kinds of Select / List boxes in the backend.
+
+        Example of data which is loaded through a data loader:
+        - Link Labels (in the inline link editor)
+        - Reference / References editor
+        - Data Sources in the Select Editor
+
+        Each Data Loader can have a slightly different API, so check the "description" field of each data loader when using it.
+
+        It is up to the data loaders to implement caching internally.
+
+        Normally, each data loader exposes the following methods:
+
+        resolveValue(options, identifier) {
+            // "options" is a DataLoader-specific object.
+            // returns Promise with [{identifier, label}, ...] list; where "identifier" was resolved to the actual object represented by "identifier".
+        }
+
+        search(options, searchTerm) {
+            // "options" is a DataLoader-specific object.
+            // returns Promise with [{identifier, label}, ...] list; these are the objects displayed in the selection dropdown.
+        }
+    `));
+
+    const dataLoadersRegistry = globalRegistry.get('dataLoaders');
+
+    dataLoadersRegistry.add('NodeLookup', {
+        description: `
+            Look up ContentRepository Nodes:
+
+            - by Node identifier (UUID) (resolveValue())
+            - by searching in node properties (search())
+
+            Takes the current context (workspace, dimensions) into account when doing the search, as this is passed in
+
+            OPTIONS:
+                - contextForNodeLinking: the current value of "selectors.UI.NodeLinking.contextForNodeLinking", required.
+                - nodeTypes: an array of node type names; if set, the search is restricted to these node types.
+        `,
+
+        _lru() {
+            if (!this._lruCache) {
```

Comment:
Hey Sebastian, could you please explain me why you use external lib insteand of es6 Map?